### PR TITLE
tony leaderboard

### PR DESCRIPTION
**Pull request by tonysun9**
Hyperparameters:
* Model (110M params)
  * 12 layers
  * 768 d_model
  * 4 * 768 d_ff
  * 12 heads
* Batch size 224
* 256 context length
* 3e-3 max lr, 1e-3 min lr
* No dropout
* 5% warmup
* GELU activation
* Gradient clip norm 1.0
* 0.01 weight decay
* 0.98 beta2

Data:
* 1.4B tokens

Took some tips from Yann:
* Mixed precision training (I used fp16), 2x speedup
* Torch compile, 2-3x speedup

Other notable experiments:
* Mixed precision with fp8 using Transformer engine
  * E4M3 forward, E5M2 backward
  * Delayed scaling
  * Initially was actually slower than fp16, need to actually change to transformer engine modules (linear, RMSNorm). Didn't use the built-in TransformerLayer module, requires dependency of FlashAttention2, which was difficult to install.
  * This surprisingly only matched the speed of fp16 (thought it would be faster due to more tensor cores for FP8 in H100s).
  * Trickier to tune, which is to be expected

Observations:
* Tweaking the hyperparameters didn't end up changing a whole lot. With the limited training time, probably better to find ways to speed up training and throw more data at the model.

### CODE REVIEW

**Comment by nelson-liu** [comment]
interesting, i thought that FP8 would be faster too. nice set of experiments!
